sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/pkversion/PartitionProcessorImpl.java

```diff
@@ -103,12 +113,16 @@ public Mono<Void> run(CancellationToken cancellationToken) {
                         Instant currentTime = Instant.now();
                         return !cancellationToken.isCancellationRequested() && currentTime.isBefore(stopTimer);
                     }).last();
-
             })
-            .flatMap(value -> this.documentClient.createDocumentChangeFeedQuery(this.settings.getCollectionSelfLink(),
-                                                                                this.options, JsonNode.class)
-                .limitRequest(1)
-            )
+            .flatMap(value -> {
+                if (throughputControlGroupConfigForFeedRange != null) {
```

Comment:
Same for this. No need for this to be in flatMap.

Reply:
This one can not, we will need to set the throughput control name for each request, because we reconstruct the options after we get the response

https://github.com/Azure/azure-sdk-for-java/blob/1c2f550b23802f10c48280b1e019c41c88a9d73f/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/epkversion/PartitionProcessorImpl.java#L143C23-L143C23